source/adapters/cuda/context.hpp

```diff
@@ -85,12 +86,47 @@ struct ur_context_handle_t_ {
 
   std::vector<ur_device_handle_t> Devices;
   std::atomic_uint32_t RefCount;
+  umf_memory_provider_handle_t memoryProviderHost = nullptr;
+  umf_memory_provider_handle_t memoryProviderDevice = nullptr;
 
   ur_context_handle_t_(const ur_device_handle_t *Devs, uint32_t NumDevices)
       : Devices{Devs, Devs + NumDevices}, RefCount{1} {
     for (auto &Dev : Devices) {
       urDeviceRetain(Dev);
     }
+
+    ur_result_t Result = UR_RESULT_SUCCESS;
+
+    ur_device_handle_t_ *device_handle = Devices[0];
+    CUdevice device0 = device_handle->get();
+    CUcontext context0 = device_handle->getNativeContext();
+
+    umf_cuda_memory_provider_params_handle_t cu_memory_provider_params =
+        nullptr;
+    umf_result_t umf_result =
+        umfCUDAMemoryProviderParamsCreate(&cu_memory_provider_params);
+    if (umf_result != UMF_RESULT_SUCCESS) {
+      Result = umf::umf2urResult(umf_result);
+      throw Result;
+    }
+
+    umf_result =
+        umf::createUmfCUDAprovider(cu_memory_provider_params, device0, context0,
```

Comment:
Please pass 0 as a device. In UMF we should check if mem type is HOST and ignore device in this case